ui/components/app/transaction-list-item/transaction-list-item.component.js

```diff
@@ -95,14 +99,24 @@ function TransactionListItemInner({
           legacy_event: true,
         },
       });
-      if (supportsEIP1559V2) {
+      if (isApprovedButNotSubmitted) {
+        dispatch(cancelTx(transactionGroup.primaryTransaction));
+      } else if (supportsEIP1559V2) {
```

Comment:
Hey @danjm : thinking of an edge case that transaction is submitted by the time the request to cancel above reaches the backend.

Reply:
@jpuri not sure I understand how this could happen. Can you please explain.